src/routes/league.ts

```diff
@@ -0,0 +1,25 @@
+import * as Router from "koa-router";
+import { Service } from "typedi";
+import { BASE_API_PREFIX } from "../util/constants";
+import { LeagueController } from "../controller/league";
+
+@Service()
+export class League {
+  constructor(private leagueController: LeagueController) {}
+
+  init(): Router {
+    const router = new Router({
+      prefix: BASE_API_PREFIX + "league"
+    });
+
+    // Routes
+    router.get("/budget/:league_id", (ctx, next) =>
+      this.leagueController.getAllMatchesSchedule(ctx, next)
+    );
+    router.get("/budget", (ctx, next) =>
+      this.leagueController.getAllMatchesSchedule(ctx, next)
+    );
```

Comment:
- base route should be `/leagues`.
- `/leagues/:id` to get the league information.
- `/leagues/:id?fields=budget`, if we want to expose only the league budget.

Examples:
- `/leagues/1` to get the league information.
- `/leagues/1?fields=buget` to get the league budget.

to get information about the latest league, we can do something similar to the following:
- `/leagues?sort=startDate&count=1`
`/leagues?fields=budget&sort=startDate&count=1`